megabus/src/main/java/com/bazaarvoice/megabus/DefaultMegabusSource.java

```diff
@@ -0,0 +1,82 @@
+package com.bazaarvoice.megabus;
+
+import com.bazaarvoice.emodb.common.uuid.TimeUUIDs;
+import com.bazaarvoice.emodb.kafka.KafkaCluster;
+import com.bazaarvoice.emodb.kafka.Topic;
+import com.bazaarvoice.emodb.sor.api.Coordinate;
+import com.bazaarvoice.megabus.guice.MegabusRefTopic;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import com.google.common.util.concurrent.Futures;
+import com.google.inject.Inject;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Clock;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation which uses the existing Megabus actors.
+ */
+public class DefaultMegabusSource implements MegabusSource {
+
+    private static final Logger _LOG = LoggerFactory.getLogger(DefaultMegabusSource.class);
+
+    private final Producer<String, JsonNode> _producer;
+    private final Topic _topic;
+    private final ObjectMapper _objectMapper;
+    private final Clock _clock;
+
+    @Inject
+    public DefaultMegabusSource(KafkaCluster kafkaCluster, @MegabusRefTopic Topic topic,
+                                ObjectMapper objectMapper, Clock clock) {
+        _producer = kafkaCluster.producer();
+        _objectMapper = objectMapper;
+        _topic = topic;
+        _clock = clock;
+    }
+
+    /*
+     * Send the given co-ordinate to the Megabus Ref Topic.
+     */
+    @Override
+    public void touch(String table, String key) {
+        // Using the minimum UUID here to make sure the time is always beyond the FCL so that the resolver is certain to put the document in to actual Megabus.
+        // This way we wouldn't be in a situation where there is a ref in Ref topic but not in the Megabus topic.
+        MegabusRef ref = new MegabusRef(table, key, TimeUUIDs.minimumUuid(), _clock.instant());
```

Comment:
do we need to add check that table exists?

Reply:
No, we should not. There will be cases where the document may not be found in that DC because of the FCL. I did some more manual testing regarding this checking all the Megabus topics including Retry and Missing ones:
1. Table doesn't exists case - The megabus Resolver will rightly  throw away the Ref from the ref topic. 
2. Document doesn't exist case - send them as NULL. This is the same case as having an older version of the EMO document in that datacenter as megabus might have a version 4 instead of the latest version 5 in the other datacenter because of FCL.   AS Long as the contract is clear and well documented (Which I will in the next commit if we decide on this path), it's fine to say that the Megabus picks the latest version available in the datacenter.